### PR TITLE
Add webUI acceptance tests for checking the versions feature

### LINKED ISSUE #33382: webUI tests for versions
Add webUI tests to check the behaviour of file versions, e.g.:
- [x] viewing versions of a file (PR #33395)
- [x] restoring versions of a file (PR #33395)
- [ ] deleting old versions of a file (I don't think you can do that in the webUI?) (PR #33585)
- [x] and for sharees to see versions of files shared with them (PR #33395)

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add webUI Acceptance test to check
- Different versions of a file are shown in the webUI
- User can restore files to earlier versions
- Different versions are shown in shared files

## Related Issue

- Fixes #33382

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Comment by dpakach** [comment]
backport #33400

**Comment by dpakach** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

File: tests/acceptance/features/webUIFiles/versions.feature [hunk 1]
```diff
@@ -0,0 +1,41 @@
+@webUI @insulated  @files_versions-app-required @comments-app-required
```
Comment (phil-davis) [inline tests/acceptance/features/webUIFiles/versions.feature]: do we really need the comments app?

File: tests/acceptance/features/webUIFiles/versions.feature [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      | user1    |
+
+  Scenario: upload new file with same name to see if different versions are shown
+    Given user "user0" has logged in using the webUI
+    And the user has browsed to the files page
+    And user "user0" has uploaded file with content "lorem content" to "/lorem.txt"
+    And user "user0" has uploaded file with content "new lorem content" to "/lorem.txt"
+    When the user browses directly to display the "versions" details of file "lorem.txt" in folder "/"
+    Then the content of file "lorem.txt" for user "user0" should be "new lorem content"
+    Then the versions list should contain 2 entries
```
Comment (phil-davis) [inline tests/acceptance/features/webUIFiles/versions.feature]: s/Then/And

File: tests/acceptance/features/webUIFiles/versions.feature [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  Scenario: sharee can see the versions of a file
+    Given user "user0" has uploaded file with content "lorem content" to "/lorem-file.txt"
+    And user "user0" has uploaded file with content "lorem" to "/lorem-file.txt"
+    And user "user0" has uploaded file with content "new lorem content" to "/lorem-file.txt"
+    And user "user0" has shared file "lorem-file.txt" with user "user1"
+    And user "user1" has logged in using the webUI
+    And the user has browsed to the files page
+    When the user browses directly to display the "versions" details of file "lorem-file.txt" in folder "/"
+    Then the content of file "lorem-file.txt" for user "user1" should be "new lorem content"
+    Then the versions list should contain 2 entries
```
Comment (phil-davis) [inline tests/acceptance/features/webUIFiles/versions.feature]: s/Then/And

File: tests/acceptance/features/lib/FilesPageElement/DetailsDialog.php [hunk 4]
```diff
@@ -113,6 +116,40 @@ private function getCommentXpath($content) {
 		return "//ul[@class='comments']//div[@class='message' and contains(., '" . $content . "')]";
 	}
 
+	/**
+	 * find the xpath of comment with given content
```
Comment (phil-davis) [inline tests/acceptance/features/lib/FilesPageElement/DetailsDialog.php]: s/comment/version

File: tests/acceptance/features/lib/FilesPageElement/DetailsDialog.php [hunk 5]
```diff
@@ -113,6 +116,40 @@ private function getCommentXpath($content) {
...
+			$versionsList,
+			__METHOD__ .
+			" could not find versions list for current file"
+		);
+		$this->waitTillElementIsNotNull($this->versionsListXpath);
+		return $versionsList;
+	}
+
+	/**
+	 * find the xpath of comment with given content
```
Comment (phil-davis) [inline tests/acceptance/features/lib/FilesPageElement/DetailsDialog.php]: s/comment/version

**Review by phil-davis — COMMENTED** [review]
Looks good, just small things - you could just add a commit to this PR  to fix those, then cherry-pick it to the back-port.

**Comment by individual-it** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Review by individual-it — APPROVED** [review]

**Pull request merged**